sql-db/hibernate-reactive/src/main/java/io/quarkus/ts/reactive/http/GroundedEndpoint.java

```diff
@@ -93,6 +101,27 @@ public GroundedEndpoint() {
                 .map(Response.ResponseBuilder::build);
     }
 
+    // https://github.com/quarkusio/quarkus/issues/18977
+    @GET
+    @Path("bookThroughSession/{authorName}/{name}")
+    public Uni<Author> createBookThroughSession(String authorName, String name) {
+        Author author = new Author();
+        author.setName(authorName);
+        return factory.withTransaction(
+                (session, transaction) -> factory.withSession(session1 -> session1.persist(author).chain(session::flush))
```

Comment:
Is opening a second session required to reproduce the bug?

Reply:
And even if we are going this way, can we, at least, have more uniformed session names? If there are three sessions, let's name them "session1, session2 and session3"(or, even better, "firstSession, secondSession and thirdSession").

Reply:
> Is opening a second session required to reproduce the bug?

yes is required. 

> f there are three sessions, let's name them "session1, session2 and session3"(or, even better, "firstSession, secondSession and thirdSession")

I prefer as it's, the suggested way is too verbose

Reply:
What about "session1" instead of "session", "session2" instead of "session1" and so on?